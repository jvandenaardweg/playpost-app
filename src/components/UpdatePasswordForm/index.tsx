import React, { useContext } from 'react';
import { KeyboardAvoidingView, Platform, View } from 'react-native';
import { Button } from '../Button';
import { UserThemeContext } from '../../contexts/UserThemeProvider';
import { InputGroupPassword } from '../InputGroup/password';
import styles from './styles';

export interface Props {
  password: string;
  isLoading: boolean;
  isSuccess: boolean;
  onChangeText(field: string, text: string): void;
  onPressUpdatePassword(): void;
}

export const UpdatePasswordForm: React.FC<Props> = React.memo((props) => {
  const { theme } = useContext(UserThemeContext);

  // Android and iOS both interact with this prop differently.
  // Android may behave better when given no behavior prop at all, whereas iOS is the opposite.
  // https://facebook.github.io/react-native/docs/keyboardavoidingview#behavior
  const behaviorOption = Platform.OS === 'ios' ? 'padding' : undefined;

  return (
    <KeyboardAvoidingView testID="UpdatePasswordForm" style={styles(theme).container} behavior={behaviorOption} keyboardVerticalOffset={60} enabled>
      <View style={styles(theme).form}>

        <InputGroupPassword
          label="Your new password"
          testID="UpdatePasswordForm-TextInput-password"
          placeholder="Your new super secret password"
          value={props.password}
          onChangeText={(text: string) => props.onChangeText('password', text)}
          onSubmitEditing={() => props.onPressUpdatePassword()}
          editable={!props.isLoading}
          returnKeyType="done"
          autoFocus
        />

        <View>
          <Button
            testID="UpdatePasswordForm-Button-update"
            title={props.isSuccess ? 'New password saved!' : 'Save new password'}
            loading={props.isLoading}
            onPress={props.onPressUpdatePassword}
            disabled={props.isLoading || props.isSuccess}
            activeOpacity={1}
          />
        </View>
      </View>
    </KeyboardAvoidingView>
  );
});
