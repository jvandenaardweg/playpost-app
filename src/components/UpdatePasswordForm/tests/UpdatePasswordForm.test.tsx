import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { UpdatePasswordForm } from '../UpdatePasswordForm';

describe('UpdatePasswordForm', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<UpdatePasswordForm onChangeText={() => {}} onPressUpdatePassword={() => {}} password="" passwordValidation="" isLoading={false} isSuccess={false} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
