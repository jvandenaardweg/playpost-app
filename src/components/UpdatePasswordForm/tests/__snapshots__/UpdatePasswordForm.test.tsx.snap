// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePasswordForm rendering should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F1F1F1",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      blurOnSubmit={false}
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Your new password"
      rejectResponderTermination={true}
      returnKeyType="done"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#e5e5e5",
          "borderRadius": 4,
          "borderWidth": 1,
          "fontSize": 15,
          "height": 50,
          "marginBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      textContentType="password"
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      blurOnSubmit={false}
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Confirm your new password"
      rejectResponderTermination={true}
      returnKeyType="done"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#e5e5e5",
          "borderRadius": 4,
          "borderWidth": 1,
          "fontSize": 15,
          "height": 50,
          "marginBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      textContentType="password"
      underlineColorAndroid="transparent"
      value=""
    />
    <View>
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#000000",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 16,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Update password
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
